feature-libs/customer-ticketing/components/list/customer-ticketing-create/customer-ticketing-create-dialog/customer-ticketing-create-dialog.component.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import {
+  AssociatedObject,
+  Category,
+  MAX_ENTRIES_FOR_ATTACHMENT,
+  TicketCreatedEvent,
+  TicketDetails,
+  TicketStarter,
+} from '@spartacus/customer-ticketing/root';
+import { FormUtils } from '@spartacus/storefront';
+import { Observable, Subscription } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { CustomerTicketingDialogComponent } from '../../../shared/customer-ticketing-dialog/customer-ticketing-dialog.component';
+@Component({
+  selector: 'cx-customer-ticketing-create-dialog',
+  templateUrl: './customer-ticketing-create-dialog.component.html',
+})
+export class CustomerTicketingCreateDialogComponent
+  extends CustomerTicketingDialogComponent
+  implements OnInit, OnDestroy
+{
+  ticketCategories$: Observable<Category[]> =
+    this.customerTicketingFacade.getTicketCategories();
+  ticketAssociatedObjects$: Observable<AssociatedObject[]> =
+    this.customerTicketingFacade.getTicketAssociatedObjects();
+  subscription: Subscription;
+
+  @Input()
+  selectedCategory: Category;
+
+  @Input()
+  selectedAssociatedObject: AssociatedObject;
+
+  attachment: File;
+
+  protected getCreateTicketPayload(form: FormGroup): TicketStarter {
+    return {
+      message: form?.get('message')?.value,
+      subject: form?.get('subject')?.value,
+      associatedTo: this.selectedAssociatedObject,
+      ticketCategory: this.selectedCategory,
```

Comment:
as mentioned, we can just do this after changing to what I mentioned

```suggestion
      ticketCategory: form?.get('ticketCategory')?.value,
```

Another point to change is to remove the parameter of the function. I don't see a reason passing the form when it's available to the `scope` of the class.